client.go

```diff
@@ -181,11 +181,28 @@ func newClient(v4 bool, v6 bool) (*client, error) {
 			log.Printf("[ERR] mdns: Failed to bind to udp6 port: %v", err)
 		}
 	}
-
 	if mconn4 == nil && mconn6 == nil {
 		return nil, fmt.Errorf("failed to bind to any multicast udp port")
 	}
 
+	// Check that unicast and multicast connections have been made for IPv4 and IPv6
+	// and disable the respective protocol if not.
+	if uconn4 == nil || mconn4 == nil {
+		log.Printf("[ERR] mdns: Failed to listen to both unicast and multicast on IPv4")
+		uconn4 = nil
+		mconn4 = nil
+		v4 = false
+	}
+	if uconn6 == nil || mconn6 == nil {
+		log.Printf("[ERR] mdns: Failed to listen to both unicast and multicast on IPv6")
```

Comment:
Same suggestion here:
```suggestion
		logger.Printf("[INFO] mdns: Failed to listen to both unicast and multicast on IPv6")
```